### ISSUE TITLE
Update development dependencies for pulumi/pulumi-provider-build-environment

### ISSUE THREAD

**Issue by im7mortal** [pulumi/pulumi-docker-containers#252]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

Should we update the development tools in  pulumi/pulumi-provider-build-environment? [specifically this section: Dockerfile](https://github.com/pulumi/pulumi-docker-containers/blob/main/docker/pulumi/Dockerfile#L129-L134)

golangci-lint to [v1.60.2](https://github.com/golangci/golangci-lint/releases/tag/v1.60.2)
pulumictl to [v0.0.46](https://github.com/pulumi/pulumictl/releases/tag/v0.0.46)

The only consideration is that Goreleaser has been updated to v2 (the current version is [v2.2.0](https://github.com/goreleaser/goreleaser/releases/tag/v2.2.0)).

**Comment by justinvp** [comment]
Yes, we should update these dependencies (and keep them up-to-date). Thanks for opening the issue. There is some concern over breaking folks (especially with the Goreleaser bump), so we'll need to message this somehow to give folks a heads-up.

**Comment by julienp** [comment]
Done with https://github.com/pulumi/pulumi-docker-containers/pull/441 and https://github.com/pulumi/pulumi-docker-containers/pull/680
